src/sentry/sentry_metrics/querying/metadata/metrics_code_locations.py

```diff
@@ -72,11 +72,15 @@ class CodeLocationsFetcher:
     # The maximum number of keys that can be fetched by the fetcher.
     #
     # The estimation was naively done by supposing at most 10 metrics with 2 projects and at most 90 timestamps.
-    MAXIMUM_KEYS = 2000
+    MAXIMUM_KEYS = 50
     # The size of the batch of keys that are fetched by endpoint.
     #
     # Batching is done via Redis pipeline and the goal is to improve the performance of the system.
-    BATCH_SIZE = 50
+    BATCH_SIZE = 25
+    # The maximum number of code locations we want to retrieve per Redis set.
+    MAX_SET_SIZE = 5
```

Comment:
`5` matches what Relay tries to enforce best effort 👍 

We could go a bit above and fetch more from Redis, sort in memory and then truncate to `5` to get a more stable list of elements, but probably overkill.

Reply:
If we assume 5 to be the "theoretical" maximum, we could do as you said and have a stable list if we use a number >= 5 but as soon as we go above that number it will become unstable. We could def do this though.

Reply:
Luckily, I saw that I already implemented sorting for consistency, so I will just increase the limit to `10`.